### PR TITLE
14057, 14065 - Improve Piece Definer image display width

### PR DESCRIPTION
**Pull request by BrentEaston**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java [hunk 1]
```diff
@@ -300,7 +304,7 @@ public void setBaseWindow(HelpWindow w) {
    * The piece defined has been changed. It may have changed size, or image
    *
    */
-  private void refresh() {
+  public void refresh() {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:307]: Why change the visibility?

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java [hunk 2]
```diff
@@ -635,7 +655,7 @@ public void mouseReleased(MouseEvent e) {
    * Record the level as current maximum and resize the image if it is not
    * at 100% already
    */
-  private void splitChanged() {
+  public void splitChanged() {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:658]: Why change the visibility?

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java (hunk 1 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:307]: Ah, That was for testing, fixing.

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java [hunk 3]
```diff
@@ -320,28 +324,43 @@ private void refresh() {
...
+      // Reset the piece to 100% scale to get the actual size.
+      final double oldScale = slot.getScale();
+      slot.setScale(1.0);
+      newSlotSize = slot.getPreferredSize();
+      slot.setScale(oldScale);
+    }
+    else {
+      slot.getComponent().revalidate();
+      slot.getComponent().repaint();
+      boolean s = slot.isValid();
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:349]: Looks unused.

**Review by uckelman — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java (hunk 3 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java]: Fixed

**Review by uckelman — APPROVED** [review]

**Pull request merged**